app/src/main/java/fr/free/nrw/commons/explore/map/ExploreMapController.java

```diff
@@ -0,0 +1,222 @@
+        return explorePlacesInfo;
+    }
+
+    /**
+     * Loads attractions from location for map view, we need to return BaseMarkerOption data type.
+     *
+     * @param curLatLng users current location
+     * @param placeList list of nearby places in Place data type
+     * @return BaseMarkerOptions list that holds nearby places
+     */
+    public static List<NearbyBaseMarker> loadAttractionsFromLocationToBaseMarkerOptions(
+        LatLng curLatLng,
+        List<ExplorePlace> placeList,
+        Context context,
+        List<Place> bookmarkLocations) {
+        List<NearbyBaseMarker> baseMarkerOptions = new ArrayList<>();
+
+        if (placeList == null) {
+            return baseMarkerOptions;
+        }
+
+        placeList = placeList.subList(0, Math.min(placeList.size(), MAX_RESULTS));
+
+        VectorDrawableCompat vectorDrawable = null;
+        VectorDrawableCompat vectorDrawableGreen = null;
+        VectorDrawableCompat vectorDrawableGrey = null;
+        VectorDrawableCompat vectorDrawableMonuments = null;
+        vectorDrawable = null;
+        try {
+            vectorDrawable = VectorDrawableCompat.create(
+                context.getResources(), R.drawable.ic_custom_map_marker, context.getTheme());
+            vectorDrawableGreen = VectorDrawableCompat.create(
+                context.getResources(), R.drawable.ic_custom_map_marker_green, context.getTheme());
+            vectorDrawableGrey = VectorDrawableCompat.create(
+                context.getResources(), R.drawable.ic_custom_map_marker_grey, context.getTheme());
+            vectorDrawableMonuments = VectorDrawableCompat
+                .create(context.getResources(), R.drawable.ic_custom_map_marker_monuments,
+                    context.getTheme());
+        } catch (Resources.NotFoundException e) {
+            // ignore when running tests.
+        }
+        if (vectorDrawable != null) {
+            Bitmap icon = UiUtils.getBitmap(vectorDrawable);
+            Bitmap iconGreen = UiUtils.getBitmap(vectorDrawableGreen);
+            Bitmap iconGrey = UiUtils.getBitmap(vectorDrawableGrey);
+            Bitmap iconMonuments = UiUtils.getBitmap(vectorDrawableMonuments);
+
+            for (ExplorePlace explorePlace : placeList) {
+                final NearbyBaseMarker nearbyBaseMarker = new NearbyBaseMarker();
+                String distance = formatDistanceBetween(curLatLng, explorePlace.location);
+                explorePlace.setDistance(distance);
+
+                nearbyBaseMarker.title(explorePlace.name);
+                nearbyBaseMarker.position(
+                    new com.mapbox.mapboxsdk.geometry.LatLng(
+                        explorePlace.location.getLatitude(),
+                        explorePlace.location.getLongitude()));
+                nearbyBaseMarker.place(explorePlace);
+                /* Help needed, Madhur another point that would be nice to have help is here where I want to create marker icons from thumbnails of places.
```

Comment:
I think you can do something like this
```
Glide.with(context)
                    .asBitmap()
                    .load(explorePlace.thumb)
                    .into(new CustomTarget<Bitmap>() {
                        @Override
                        public void onResourceReady(@NonNull Bitmap resource, @Nullable Transition<? super Bitmap> transition) {
                            nearbyBaseMarker.setIcon(IconFactory.getInstance(context).fromBitmap(resource));
                        }

                        @Override
                        public void onLoadCleared(@Nullable Drawable placeholder) {
                        }
                    });
```

Reply:
Ah I see you already tried this, not sure why it isn't working, can you confirm if `onResourceReady` is called?